pythonforandroid/toolchain.py

```diff
@@ -141,6 +141,9 @@ def wrapper_func(self, args):
                                       user_ndk_ver=self.ndk_version,
                                       user_ndk_api=self.ndk_api)
         dist = self._dist
+        bs = Bootstrap.get_bootstrap(args.bootstrap, ctx)
+        # recipes rarely change, but during dev, bootstraps can change from
+        # build to build, so run prepare_bootstrap even if needs_build is false
```

Comment:
The comment says we run `prepare_bootstrap()` but I don't see that we do :confused: 
Also why are we calling `Bootstrap.get_bootstrap()` for?

Reply:
Interesting! Might be an oversight by @kollivier and maybe it was supposed to have a `ctx.prepare_bootstrap(bs)` right after? That would explain why `bs` is unused and there's the confusing comment there. I don't reuse any of the build layout for subsequent builds, so I assume that's why I didn't run into a situation in my tests where this was relevant

Reply:
The code wasn't used in the original patch, either, so whatever I was intending to do, I never actually made it happen. I think it can just be removed. 

From the comment, I suspect that during some tests I found a case where switching the bootstrap didn't trigger a rebuild. Even if that was still true, it's likely not a common case (I was triggering it no doubt because I was testing with different bootstraps!) and can be fixed with a clean rebuild.

Reply:
Also, thanks for cleaning this up and getting it running against the latest code! :)

Reply:
@kollivier you're welcome, and this would have been infinitely more work without your foundation, so thanks again for starting this!